pkg/processor/sourceprocessor/attribute_filler.go

```diff
@@ -113,11 +113,11 @@ func (f *attributeFiller) fillAttributes(atts *pdata.AttributeMap) bool {
 func (f *attributeFiller) resourceLabelValues(atts *pdata.AttributeMap) []interface{} {
 	arr := make([]interface{}, 0)
 	for _, label := range f.labels {
-		value, ok := atts.Get(label)
-		if !ok {
-			return nil
+		if value, found := atts.Get(label); found {
+			arr = append(arr, value.StringVal())
+		} else {
+			arr = append(arr, "undefined")
```

Comment:
Define as `const`?

Reply:
This value is only being used once - unless you count the appearances in tests. Do you think I should replace the "undefined"-s in tests with the const as well? I'm on the fence about it .

Reply:
🤷  Let's leave it then. I don't have a strong opinion about it but I'm worried about the number of occurrences of this magic string growing because X.

> ... I'm on the fence about it.

Then jump :)